### PR TITLE
docs: Textual improvements in build docs

### PR DESCRIPTION
**Pull request by merland**
While reading the build docs, I found some opportunities for textual improvements (Force of habit, I used to work as a technical writer...)   

* Added a few missing words, should be uncontroversial. 

* Changed/added some punctuation, for better flow and readability. 

* Fixed one Markdown issue, where two list item headings rendered without a line break. (See image)
This one needs to be verified after a build, I don't have a proper build environment yet.   

layout_issue

### CODE REVIEW

**Comment by promag** [comment]
ACK, these look like reasonable changes, just squash commits. Also, please go easy on PR's like this, like wait until there's a reasonable amount of improvements before creating the pull request.

File: doc/build-unix.md [hunk 1]
```diff
@@ -194,7 +194,7 @@ Hardening Flags:
 
 Hardening enables the following features:
 
-* Position Independent Executable
+* _Position Independent Executable_
```
Comment (promag) [inline doc/build-unix.md]: Remove trailing whitespaces.

Comment (merland) [reply doc/build-unix.md]: Well, that is the main change, double-space causes a linebreak: https://markdown-guide.readthedocs.io/en/latest/basics.html#line-return 

Is there a preferred way of doing linebreaks in this context?

**Comment by merland** [comment]
@promag Thanks for reviewing. 
Could you please elaborate on why we should go easy on PRs like this? Where can I read more about this rationale? I'm trying to learn how to best contribute. 

Small PRs are nice since the conflict risk is lower, but of course they increase the administration overhead. What is the best balance?  

I left it un-squashed so that it would be easier to comment/discuss the individual commits. Was planning to squash right before merge.

File: doc/build-unix.md (hunk 1 above)
Comment (promag) [reply doc/build-unix.md]: Either change formatting, update the failing `test/lint/lint-whitespace.sh` or leave as it is.

**Review by promag — COMMENTED** [review]
> What is the best balance?

I'd say just try to **not** have more than 1 open PR with these improvements and wait a couple of days /weeks before opening a new one. There is for sure room for a lot of textual improvements and we don't want that to be the focus of the maintainers. Please read https://github.com/bitcoin/bitcoin/blob/master/.github/PULL_REQUEST_TEMPLATE.md.

File: doc/build-unix.md (hunk 1 above)
Comment (merland) [reply doc/build-unix.md]: Ah, didn't notice the failed check, sorry. I'll figure something out.

**Comment by merland** [comment]
@promag 
I have read the PR template before, but it does not mention small textual improvement PRs. Your suggestion of having only one of this kind of PRs open at time sounds reasonable, but it seems that the governance structure of the project puts a break on "trivial" changes being submitted and accepted. This is a pity, I think.  
True excellence can only be achieved if we fix all broken windows, even the tiny ones.

**Comment by practicalswift** [comment]
@merland 

This is my perspective:

I'm one of the reviewers who will review your documentation changes.

I got really happy when reading your PR: having a former technical writer looking over and improving our documentation is a great opportunity. Excellent news!

Don't hesitate to ping me if you ever need a documentation PR reviewed.

In the highly unlikely event that you make "too many" high quality documentation improvements we'll have to solve that by trying to expand the pool of documentation reviewers.

(This might be highly controversial but personally I'm not convinced that PR count is a good proxy for review burden. Personally I feel that review burden is largely a function of LOC touched. Documentation changes are typically tiny in terms of LOC touched.)

Thanks for the contribution! I'm really happy to have you on board :-)

ACK 2b3e0a18fd193b64d01904cf1f6588947b9a41a6 modulo squash

> True excellence can only be achieved if we fix all broken windows, even the tiny ones.

Agreed!

File: doc/build-osx.md [hunk 2]
```diff
@@ -20,23 +20,23 @@ Dependencies
 
 See [dependencies.md](dependencies.md) for a complete overview.
 
-If you want to build the disk image with `make deploy` (.dmg / optional), you need RSVG
+If you want to build the disk image with `make deploy` (.dmg / optional), you need RSVG.
```
Comment (hebasto) [inline doc/build-osx.md]: `If you want to build the disk image with `make deploy` (.dmg / optional), you need RSVG:`

File: doc/build-osx.md [hunk 3]
```diff
@@ -20,23 +20,23 @@ Dependencies
...
+like so:
 
 ```shell
 ./contrib/install_db4.sh .
 ```
 
 from the root of the repository.
 
-**Note**: You only need Berkeley DB if the wallet is enabled (see the section *Disable-Wallet mode* below).
+**Note**: You only need Berkeley DB if the wallet is enabled (see the section *Disable-wallet mode* below).
```
Comment (hebasto) [inline doc/build-osx.md]: **Note**: You only need Berkeley DB if the wallet is enabled (see the section [*Disable-wallet mode*](/doc/build-osx.md#disable-wallet-mode) below).

**Comment by hebasto** [comment]
ACK 6202fc4  with some nits:
- could references to own sections (e.g., "Disable-wallet mode" section) be clickable and formatted consistently;
- some more lines need `:` at the end;
- could be better to replace "Clone the Bitcoin Core source code and cd into bitcoin" with just "Clone the Bitcoin Core source code:"

File: doc/build-osx.md (hunk 3 above)
Comment (hebasto) [reply doc/build-osx.md]: Make section name as a link.

**Comment by merland** [comment]
@hebasto Thanks for reviewing. 
I think I agree to both of your last comments, but I want to push this PR through before it grows even more. I will make a note of your suggestions and then any of us can create a new PR further on.

**Comment by merland** [comment]
Can someone tell me why the [AppVeyor build is failing](https://ci.appveyor.com/project/DrahtBot/bitcoin/builds/19840684)? Seems unrelated...

**Comment by merland** [comment]
@hebasto I reconsidered and have pushed almost all of your suggestions. Thanks.

**Comment by merland** [comment]
Squashed to 1 commit

**Comment by fanquake** [comment]
utACK c68464f

The changes look fine, however, please update the commit description so that it doesn't include all the messages from the interim commits, and remove any `@` mentions (they just lead to spam mentions/emails when downstream projects pull our commits).

> Was planning to squash right before merge.

Please don't open PRs with the intent to "cleanup" and squash after review/before merge, as that just means another round of review (less of an issue here, but will always be the case for code changes).

PR's should either be clearly marked [wip] if there is technical and/or code feedback required (that couldn't be discussed in an issue), otherwise, if you open a PR it should be in a ready to merge state.

**Comment by merland** [comment]
@fanquake Thanks a lot for clearing this up for me. I'm new to both squashing and the project workflow, so this was a good learning experience. 

The commit message has been cleaned up now.

**Comment by fanquake** [comment]
re-utACK 36c8e68

**Comment by hebasto** [comment]
re-utACK 36c8e68585091d10d8a1d4f97d2db87cd8cedcb2

**Pull request merged**